### PR TITLE
Bump MEGAHIT to 1.2.6

### PR DESCRIPTION
**Pull request by apcamargo**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by apcamargo** [comment]
Hey @bioconda/core 
This is my first recipe of a non-Python/R package, so this may be a beginner's mistake. The build fails because it doesn't find `CMakeLists.txt`, even though the file is in the root of the [downloaded `.tar.gz` file](https://github.com/voutcn/megahit). What should I do here? Is it necessary to change directory?

File: recipes/megahit/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,23 @@
-{% set version = "1.1.3" %}
-{% set sha256 = "b6eefdee075aaf7a8f9090e2e8b08b770caff90aa43a255e0e220d82ce71c492" %}
+{% set version = "1.2.5-beta" %}
```
Comment (dpryan79) [inline recipes/megahit/meta.yaml]: As a rule we don't ship beta versions.

Comment (apcamargo) [reply recipes/megahit/meta.yaml]: Sorry for overlooking this. In any case, I did some progressions in this recipe and I'll use it when the final version is ready.

**Comment by apcamargo** [comment]
Hey @bioconda/osx ! Any clue of what is happening here? The recipe was building fine until yesterday.

**Comment by mbargull** [comment]
> The recipe was building fine until yesterday.

Could you point me to the log of the last successful build, for reference?

**Comment by apcamargo** [comment]
@mbargull sure!

https://github.com/bioconda/bioconda-recipes/runs/168622577

**Comment by mbargull** [comment]
Thanks. We updated the `conda-build` version in our build setup yesterday, so I checked whether this caused any change in the list of packages used for the build. The only difference is an updated `bzip2` package (that change is independent of our `conda-build` update as that package has been updated yesterday, too). But I couldn't see any changes that may be relevant...
I'm clueless at the moment why we are getting
```
21:09:43 BIOCONDA INFO (OUT) -- Check for working C compiler: $BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang -- broken
```
here, sorry.

**Comment by mbargull** [comment]
If anyone is more familiar with building under macOS, the error we get is `ld: library not found for -lSystem`:
```
04:58:03 BIOCONDA INFO (ERR) CMake Error at /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_build_env/share/cmake-3.14/Modules/CMakeTestCCompiler.cmake:60 (message):
04:58:03 BIOCONDA INFO (ERR)   The C compiler
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)     "/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_build_env/bin/x86_64-apple-darwin13.4.0-clang"
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)   is not able to compile a simple test program.
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)   It fails with the following output:
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)     Change Dir: /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/work/build/CMakeFiles/CMakeTmp
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)     Run Build Command(s):/usr/bin/make cmTC_d9370/fast
04:58:03 BIOCONDA INFO (ERR)     /Applications/Xcode-8.3.3.app/Contents/Developer/usr/bin/make -f CMakeFiles/cmTC_d9370.dir/build.make CMakeFiles/cmTC_d9370.dir/build
04:58:03 BIOCONDA INFO (ERR)     Building C object CMakeFiles/cmTC_d9370.dir/testCCompiler.c.o
04:58:03 BIOCONDA INFO (ERR)     /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_build_env/bin/x86_64-apple-darwin13.4.0-clang   -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -I/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include -fdebug-prefix-map=/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/work=/usr/local/src/conda/megahit-1.2.6 -fdebug-prefix-map=/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p=/usr/local/src/conda-prefix  -isysroot /Applications/Xcode-8.3.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.9   -o CMakeFiles/cmTC_d9370.dir/testCCompiler.c.o   -c /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/work/build/CMakeFiles/CMakeTmp/testCCompiler.c
04:58:03 BIOCONDA INFO (ERR)     Linking C executable cmTC_d9370
04:58:03 BIOCONDA INFO (ERR)     /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_build_env/bin/cmake -E cmake_link_script CMakeFiles/cmTC_d9370.dir/link.txt --verbose=1
04:58:03 BIOCONDA INFO (ERR)     /Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_build_env/bin/x86_64-apple-darwin13.4.0-clang -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -I/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include -fdebug-prefix-map=/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/work=/usr/local/src/conda/megahit-1.2.6 -fdebug-prefix-map=/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p=/usr/local/src/conda-prefix  -isysroot /Applications/Xcode-8.3.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.9 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib -L/Users/distiller/project/miniconda/conda-bld/megahit_1563278142070/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib  CMakeFiles/cmTC_d9370.dir/testCCompiler.c.o  -o cmTC_d9370
04:58:03 BIOCONDA INFO (ERR)     ld: library not found for -lSystem
04:58:03 BIOCONDA INFO (ERR)     clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
04:58:03 BIOCONDA INFO (ERR)     make[1]: *** [cmTC_d9370] Error 1
04:58:03 BIOCONDA INFO (ERR)     make: *** [cmTC_d9370/fast] Error 2
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR)   CMake will not be able to correctly generate this project.
04:58:03 BIOCONDA INFO (ERR) Call Stack (most recent call first):
04:58:03 BIOCONDA INFO (ERR)   CMakeLists.txt:2 (project)
04:58:03 BIOCONDA INFO (ERR) 
04:58:03 BIOCONDA INFO (ERR) 
```

**Comment by mbargull** [comment]
The failure is most likely due to the same reason as https://github.com/bioconda/bioconda-recipes/pull/16402#issuecomment-511840128 . We'll fix/workaround this in `bioconda-utils` soon.

**Comment by apcamargo** [comment]
Thank you for the fix @mbargull !

I think the recipe is now ready to merge.

**Pull request merged**